src/client/usecase.ts

```diff
@@ -76,6 +77,13 @@ class UseCaseBase implements Interceptable {
     this.metadata.provider = providerConfig.name;
     hookRouter.setCurrentProvider(providerConfig.name);
 
+    //If we have security values pass them directly to perform
+    if (options?.security) {
+      providerConfig = new ProviderConfiguration(
```

Comment:
suggestion: once again, I would extract this to a function `const providerConfig = getProviderConfiguration(options)`